### PR TITLE
if nothing to build, do not push

### PR DESCRIPTION
**Pull request by deitch**
Signed-off-by: Avi Deitcher <avi@deitcher.net>

**- What I did**

With the earlier PR #3863 , for `pkg build`, if the image exists in the remote registry we don't bother pulling it unless they explicitly requested `--pull`. This is much more efficient.

However, this might break `pkg push` in the following scenario:

1. `lkt pkg push pkg/foo`
2. The image is not in cache at all, but is entirely in registry
3. Nothing gets built (correctly), nothing gets pulled (correctly)
4. It now tries to push from cache and finds nothing there.
5. This fixes it so that it does not try to push if it did not build anything or find anything local

**- How I did it**

Added logic to check if we built anything or found anything locally.

I would add that there might be a cleaner way to do this, specifically just check before pushing, "do we have anything in cache". The net result, though, is the same.

**- How to verify it**

CI

**- Description for the changelog**

Handle `pkg push` when nothing is built or pulled.

### CODE REVIEW

File: src/cmd/linuxkit/pkglib/build.go [hunk 1]
```diff
@@ -248,34 +253,23 @@ func (p Pkg) Build(bos ...BuildOpt) error {
 		// check local cache, fallback to check registry / pull image from registry, fallback to build
 		fmt.Fprintf(writer, "checking for %s in local cache...\n", ref)
 		for _, platform := range bo.platforms {
-			if exists, err := c.ImageInCache(&ref, "", platform.Architecture); err == nil && exists {
+			exists, err := c.ImageInCache(&ref, "", platform.Architecture)
+			switch {
+			case err != nil && exists:
```
Comment (giggsoff) [inline src/cmd/linuxkit/pkglib/build.go:258]: Should it be `case err == nil && exists:` as before?

Comment (deitch) [reply src/cmd/linuxkit/pkglib/build.go:258]: Yeah, just maybe.

Comment (deitch) [reply src/cmd/linuxkit/pkglib/build.go]: Fixed. Thanks for finding.

**Review by giggsoff — APPROVED** [review]
It solves the problems in CI of EVE-OS (at least on my fork with updated linuxkit and removal of additional pull steps before push).

**Pull request merged**